project.clj

```diff
@@ -24,7 +24,7 @@
 
   :test-paths ["test/common"] ;; See `test-clj` and `test-cljs` profiles below.
 
-  :profiles {:provided {:dependencies [[org.clojure/clojure "1.7.0"]]}
+  :profiles {:provided {:dependencies [[org.clojure/clojure "1.9.0-alpha10"]]}
```

Comment:
I think it's not that much about leiningen. The project.clj should just keep saying `1.7.0`. The tricky part is requiring the spec namespace. We need a separate namespace in cider-nrepl that provides the spec functions we use, and then this namespace takes care to only require `clojure.spec` if it is available.